### ISSUE TITLE
Support buildkit build without dependency on kubeconfig

### ISSUE THREAD

**Issue by markphillips100** [loft-sh/devspace#2063]
**Is your feature request related to a problem?**
I am attempting to use devspace in my Azure Pipelines CI for image builds rather than the usual sequential pipeline task, one per image.  Using devspace would allow me to take advantage of parallel builds and the pipelines for creating a single base image for all other images to use.

I'm having a similar problem to #1769  only the related merged PR doesn't solve it for me as I do use buildkit in order to use `--secrets` to support passing of a token to the docker build - necessary for the purpose of accessing a private package feed (in Azure).

So basically I need buildkit enabled and have no kubernetes cluster so no kubeconfig and no namespace.

**Which solution do you suggest?**
Would it be possible to introduce a flag to prevent the `devspace build` or `build_images` in the pipeline needing the kubeconfig rather than relying on if buildkit is used or not?

**Which alternative solutions exist?**
None for me.  I need buildkit if I'm to use my yaml in CI for image builds.

**Additional context**

/kind feature

**Comment by markphillips100** [comment]
I should add that I'm focused (fixated) on the v6 alpha versions :-) due to the pipelines feature, so looking for something to hopefully help in v6.

**Comment by FabianKramm** [comment]
@markphillips100 thanks for creating this issue! We'll see what we can do there, but generally this should be possible

**Comment by markphillips100** [comment]
Thank you @FabianKramm .  I really appreciate it.

**Comment by markphillips100** [comment]
@FabianKramm do you have any expectation on a possible solution for this issue?  No pressure, just wanted to do some planning of my own is all.

**Comment by markphillips100** [comment]
Any updates, or whether this change would make it into v6 @FabianKramm ?

**Comment by FabianKramm** [comment]
@markphillips100 I believe this should work with the latest DevSpace version. Is it currently still a problem?

**Comment by markphillips100** [comment]
Yep all good.  Just shows a warning about a missing kubeconfig but no longer prevents the image builds.  Thank you @FabianKramm

**Comment by alexandradragodan** [comment]
Hello! 
Thank you @markphillips100 for raising this issue up! 

We're glad your use case was successful with our updated version. So, I will close this issue as there are no clarifications left. 

If you have any further questions or concerns, we'd be happy to support!
